### PR TITLE
fix: QualifiedIdMapper crashes [AR-2137]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2137" title="AR-2137" target="_blank">BugAR-2137</a>  App is crashing when tapping on "New Conversation" Button
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

QualifiedIdMapper is only provided in a ViewModel scope and requires user session. It's wrongly injected into BroadcastReceivers - they must have zero arguments constructor to work so everything needs to be injected as `lateinit var`. 

### Solutions

Provided another QualifiedIdMapper with @NoSession annotation in a SingletonComponent and injected with this annotation into @AndroidEntryPoint BroadcastReceivers (they doesn't have its own component so they use SingletonComponent) using `lateinit var`  to bring back empty constructors.
Restore conversationId to be nullable again in OtherUserProfileScreenViewModel.

### Dependencies (Optional)

https://github.com/wireapp/kalium/pull/753

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

Previously it crashed when opening login screen, add conversation screen, other user profile screen (from any other place than group conversation) or when swiping out notifications - it shouldn't now.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-824/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4056942)(4056942)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4056942)(4056942)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/notification/broadcastreceivers/ConnectionRequestNotificationDismissReceiver.kt [hunk 1]
```diff
@@ -49,14 +51,7 @@ class ConnectionRequestNotificationDismissReceiver(
                 }
 
             sessionScope?.let { scope ->
-                requesterUserId?.let { userId ->
-                    qualifiedIdMapper.fromStringToQualifiedID(userId).let {
-                        scope.conversations.markConnectionRequestAsNotified(it)
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/notification/broadcastreceivers/ConnectionRequestNotificationDismissReceiver.kt:54]: why aren't we marking connection request as notified anymore? 🤔

**Review by alexandreferris — COMMENTED** [review]
Great work @saleniuk !
I just have one small question

File: app/src/main/kotlin/com/wire/android/notification/broadcastreceivers/ConnectionRequestNotificationDismissReceiver.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/notification/broadcastreceivers/ConnectionRequestNotificationDismissReceiver.kt:54]: We do, it was just duplicated by mistake 😄 
image

Comment (alexandreferris) [reply app/src/main/kotlin/com/wire/android/notification/broadcastreceivers/ConnectionRequestNotificationDismissReceiver.kt:54]: ooooooh, didn't realize that! 😓 my bad

**Review by alexandreferris — APPROVED** [review]

**Pull request merged**